launch_ros/launch_ros/utilities/to_parameters_list.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module with utility to transform evaluated parameters into parameter lists."""
+
+import pathlib
+from typing import Dict
+from typing import List
+
+import rclpy.parameter
+import yaml
+
+from ..parameters_type import EvaluatedParameters
+from ..parameters_type import EvaluatedParameterValue  # noqa
+
+
+def to_parameters_list(
+    evaluated_parameters: EvaluatedParameters
+) -> List[rclpy.parameter.Parameter]:
+    """
+    Transform evaluated parameters into a list of rclpy.parameter.Parameter objects.
+
+    :param parameters: parameters as either paths to parameter files or name/value pairs.
+        See `evaluate_parameters()` documentation for further reference.
+    :returns: a list of parameters
+    """
+    parameters = []  # type: List[rclpy.parameter.Parameter]
+    for params_set_or_path in evaluated_parameters:
+        if isinstance(params_set_or_path, pathlib.Path):
+            with open(str(params_set_or_path), 'r') as f:
+                params_set = yaml.load(f)  # type: Dict[str, EvaluatedParameterValue]
```

Comment:
I think if loaded from a yaml file there can be nested dictionaries, so the name should be joined with a `.` from each level. A hack might be to call `normalize_parameters/evaluate_parameters` on the loaded yaml to flatten it

Reply:
Hmm, that's a very good point. I'll flatten it.